### PR TITLE
 portenta h7 m7 ethernet support

### PR DESCRIPTION
**Pull request by Jenifen**
add native Ethernet support for portenta h7 m7 machine control

### CODE REVIEW

File: src/micro_ros_arduino.h [hunk 1]
```diff
@@ -95,11 +102,12 @@ extern "C" bool arduino_wifi_transport_open(struct uxrCustomTransport * transpor
 extern "C" bool arduino_wifi_transport_close(struct uxrCustomTransport * transport);
 extern "C" size_t arduino_wifi_transport_write(struct uxrCustomTransport* transport, const uint8_t * buf, size_t len, uint8_t * err);
 extern "C" size_t arduino_wifi_transport_read(struct uxrCustomTransport* transport, uint8_t* buf, size_t len, int timeout, uint8_t* err);
-
+#if not defined(TARGET_PORTENTA_H7_M7)
```
Comment (pablogs9) [inline src/micro_ros_arduino.h:105]: Use `#ifndef`

File: src/micro_ros_arduino.h [hunk 2]
```diff
@@ -48,7 +48,14 @@ static inline void set_microros_transports(){
 #include <NativeEthernet.h>
 #endif
 
-#if defined(TARGET_STM32F4) || defined(ARDUINO_TEENSY41)
+#if defined(TARGET_PORTENTA_H7_M7) 
+#include <EthernetUdp.h>
+#include <PortentaEthernet.h>
+#include <uxr/client/transport.h>
```
Comment (pablogs9) [inline src/micro_ros_arduino.h:54]: Why do we need this include?

File: examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+#include <stdio.h>
+#include <rcl/rcl.h>
+#include <rcl/error_handling.h>
+#include <rclc/rclc.h>
+#include <rclc/executor.h>
+#include <std_msgs/msg/int32.h>
+
+#if !defined(TARGET_STM32F4) || !defined(ARDUINO_TEENSY41) || !defined(TARGET_PORTENTA_H7_M7)
+#error This example is only avaible for Arduino Portenta, Arduino Teensy41 and STM32F4
```
Comment (pablogs9) [inline examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino:11]: Has this been tested in F4 and Teensy 4.1?

File: examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+{
+  RCLC_UNUSED(last_call_time);
+  if (timer != NULL) {
+    RCSOFTCHECK(rcl_publish(&publisher, &msg, NULL));
+    msg.data++;
+  }
+}
+
+void setup() {
+  byte arduino_mac[] = { 0xDE, 0xAD, 0xBE, 0xEF, 0xFE, 0xED };
```
Comment (pablogs9) [inline examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino:44]:
```suggestion
  byte arduino_mac[] = { 0xAA, 0xBB, 0xCC, 0xEE, 0xDD, 0xFF };
```

File: examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+  // The IP address will be dependent on your local network
+  IPAddress arduino_ip(192, 168, 1, 177);
+  IPAddress agent_ip(192, 168, 1, 113);
```
Comment (pablogs9) [inline examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino:47]:
```suggestion
  IPAddress arduino_ip(192, 168, 1, 177);
  IPAddress agent_ip(192, 168, 1, 113);
```

**Review by pablogs9 — CHANGES_REQUESTED** [review]
Minor comments

File: src/micro_ros_arduino.h (hunk 2 above)
Comment (Jenifen) [reply src/micro_ros_arduino.h:54]: okay, we need just this header PortentaEthernet.h, i'll change it

File: examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino (hunk 3 above)
Comment (pablogs9) [reply examples/micro-ros_publisher_ethernet/micro-ros_publisher_ethernet.ino:11]: Should be fixed

**Comment by pablogs9** [comment]
@mergify backport main foxy

**Pull request merged**